frontend/src/metabase/admin/settings/components/SettingsBatchForm.jsx

```diff
@@ -40,6 +40,7 @@ export default class SettingsBatchForm extends Component {
   constructor(props, context) {
     super(props, context);
     this.state = {
+      pristine: true,
```

Comment:
what's the difference between `pristine` and `dirty` and is there a way to avoid `pristine` and just use `dirty` in this component? I'm not sure what `pristine: true, dirty: true` or `pristine: false, dirty: falsy` would mean

Reply:
`pristine: false` means that that a user has changed value. So if a user changed a value and then reverted it back, the form will be pristine so it makes not much sense to allow to submit the form. Do you think handling this case worth it here?

Reply:
I'm wondering if we can merge the logic of `pristine` and `dirty` into a single boolean since they sort of represent the same thing; they're the inverse of each other.

The logic around setting `dirty` is more clumsy. See where you are iterating through each `value` to check for equality for the `pristine` value but `dirty` is blindly set to `true` in the `handleChangeEvent` callback:

```
      const pristine = this.props.elements.every(
        ({ key, value }) => settingsValues[key] === value,
      );

      return {
        pristine,
        dirty: true,
```

It seems like you could just make `dirty` a little smarty here and then use it to determine the `disabled` prop

Reply:
Yes, thanks, it is possible to use just one flag here since we do not have validation that requires `dirty` state. It is enough to use only `pristine` which means that the values in inputs are the same as initial ones even though inputs were changed by a user and reverted back. Just wanted to keep the naming and behavior of these flags to be the same as in most of the form libraries. Thanks for noticing!